core/tests/performance_framework/process_data.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Script to process page loading stats or HTTP Archive(also referred to as HAR)
+data and/or timing stats to obtain stats/metrics for the corresponding page.
+
+We need selenium + browsermob-proxy to capture page loading session
+(or HTTP Archive), to obtain page size stats. However, owing to the sluggish
+nature of the proxy server, timing stats are not accurate.
+Hence, we have separate timing stats which are obtained without making
+use of the proxy, directly from the browser console.
+"""
+
+
+class PageSessionMetricsRetriever(object):
+    """Contains methods to process stats and provide performance metrics.
+
+    page_session_stats if a dictionary containing page load statistics or
+    HTTP Archive (HAR).
+    Refer: https://dvcs.w3.org/hg/webperf/raw-file/tip/specs/HAR/Overview.html
+
+    page_session_timings is a dictionary containing various timings associated
+    with a page load like request time, dom load time, etc.
+    Structure:
```

Comment:
We generally describe the structure of a Python object as such:

```
page_session_timings is a dictionary containing metrics associated with page
loading and includes the following keys:

    webkitClearResourceTimings: maps to a dict containing (please finish)
    memory: maps to a dict containing (please finish)
    timing: maps to a dict containing the keys:
        domLoading: the Unix time the DOM finished loading (in milliseconds) (please ensure this is right)
        fetchStart: the Unix time the page began loading (in milliseconds) (please ensure this is right)
```

Another example in the code base, for reference: https://github.com/oppia/oppia/blob/develop/core/domain/collection_services.py#L72